### PR TITLE
Add cmdlets for ADL deleted items seach and restore

### PR DESCRIPTION
**Pull request by akaguptams**
## Description

Add cmdlets for ADL deleted items enumerate and restore
Design: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/227

## Checklist

- [ Yes ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ Yes ] The title of the PR is clear and informative
- [ Yes ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ Yes ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ Yes ] If applicable, the changes made in the PR have proper test coverage
- [ Yes ] For public API changes to cmdlets:
    - [ Yes ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by maddieclayton** [comment]
@azuresdkci Add to whitelist

**Comment by maddieclayton** [comment]
@akaguptams To get the tests to pass, please be sure to copy your newly recorded tests from DataLakeStore.Test/bin/Debug/netcoreapp2.1/SessionRecords into DataLakeStore.Test/SessionRecords.  Right now they seem to be missing from this PR.

**Comment by akaguptams** [comment]
@maddieclayton I have added the json files now from the recorded tests.

**Comment by maddieclayton** [comment]
@akaguptams Please fix help validation errors: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell/10736/artifact/artifacts/ValidateHelpIssues.csv/*view*/

File: src/DataLakeStore/DataLakeStore/help/Restore-AzDataLakeStoreDeletedItem.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
+﻿---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.DataLakeStore.dll-Help.xml
+Module Name: Az.DataLakeStore
+ms.assetid: D231E9A0-DC1E-411B-A87A-56A8C767F6C5
+online version: https://docs.microsoft.com/en-us/powershell/module/az.datalakestore/restore-azdatalakestoredeleteditem
+schema: 2.0.0
```
Comment (maddieclayton) [inline src/DataLakeStore/DataLakeStore/help/Restore-AzDataLakeStoreDeletedItem.md]: Please regenerate this file using https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md. This should fix your CI failure.

**Comment by akaguptams** [comment]
@maddieclayton I have fixed the help files.

**Pull request merged**